docs/GettingStartedDocs/best_practices_secure_enclaves.md

```diff
@@ -0,0 +1,275 @@
+
+> It's important to note that enclaves do not provide _code confidentiality_. This is noteworthy in the context of this discussion because, as will be discussed, hard-coded credentials in application code are to be avoided -- with or without enclave protections.
+
+Enclaves provide new trust boundary protections that address old threats and open new capabilities. But just as Open Enclave and TEEs shift the security model, application developers must shift how they handle data, especially secrets, like keys and passwords.
+<br />
+<br />
+<br />
+
+## 2.1 How _Not_ to Handle Application Secrets
+Let's remind ourselves of a way _not_ to handle application secrets. A common example of this well-discussed application weakness is described by [CWE-798: Use of Hard-coded Credentials (4.4) (mitre.org)](https://cwe.mitre.org/data/definitions/798.html). While application security is often a tradeoff between multiple factors and broad edicts are not always appropriate, it's not controversial to simply say that hard-coding credentials in application code _should not be done_.
+<br />
+<br />
+<br />
+
+## 2.2 The New, Secure Enclave Way to Handle Application Secrets
+
+Enclaves provide two properties that enable applications to handle secrets securely: 1) strong identity, 2) runtime protection of secrets in memory. These two properties enable secure point-to-point secret transfers. When applications are built with enclaves, all secret management should occur dynamically _inside_ the enclave and through secure communications with trusted sources, such as cloud provider key vaults. Let's look at one example of how enclave attestation and enclave memory protection can combine to implement a secure channel.
+
+### Secure Key Release
+One of the core challenges customers have interacting with encrypted environments is how to ensure that they can securely and reliably communicate with the code running in the environment ("enclave code").
+
+One solution to this problem is what is known as "Secure Key Release", which is a pattern that enables this kind of communication with enclave code.
+
+To implement the "Secure Key Release" pattern, the enclave code generates an ephemeral asymmetric key. It then serializes the public portion of the key to some format (possibly a JSON Web Key, or PEM, or some other serialization format).
+The enclave code then calculates the SHA256 value of the public key and passes it as an input to code which generates an SGX Quote. For Open Enclave, this is done with `oe_get_evidence` or `oe_get_report`.
```

Comment:
```suggestion
The enclave code then calculates a hash (e.g., SHA256) value of the public key and passes it as an input to code which generates attestation evidence. For Open Enclave, this can be done with `oe_get_evidence`.
```
Avoid SGX-specific terminology and references to deprecated/removed APIs.

Reply:
I took your edits, verbatim.